src/pretix/static/pretixcontrol/js/ui/main.js

```diff
@@ -6,6 +6,44 @@ function question_page_toggle_view() {
 
     show = $("#id_type").val() == "B" && $("#id_required").prop("checked");
     $(".alert-required-boolean").toggle(show);
+
+    let input = $('#id_default_value');
+    let common_attrs = ' name="default_value" placeholder="' + input.attr('placeholder') + '" title="' + input.attr('title') + '" id="id_default_value"';
+    let value = input.val();
+    switch ($("#id_type").val()) {
+        case 'N':
+            input.replaceWith('<input type="number" class="form-control" value="' + value + '" ' + common_attrs + '>');
+            $('.form-group:has(#id_default_value)').show();
+            break;
+        case 'S':
+            input.replaceWith('<input type="text" maxlength="190" class="form-control" value="' + value + '" ' + common_attrs + '>');
+            $('.form-group:has(#id_default_value)').show();
+            break;
+        case 'T':
+            input.replaceWith('<textarea cols="40" rows="10" class="form-control" ' + common_attrs + '>' + value + '</textarea>');
+            $('.form-group:has(#id_default_value)').show();
+            break;
+        case 'B':
+            let checked = (value === 'True' || value === 'on' ? 'checked' : '');
+            input.replaceWith('<input type="checkbox" ' + common_attrs + ' ' + checked + '>');
+            $('.form-group:has(#id_default_value)').show();
+            break;
+        case 'F':
+            $('.form-group:has(#id_default_value)').hide();
+            break;
+        case 'D':
+            input.replaceWith('<input type="date" class="form-control" value="' + value + '" ' + common_attrs + '>');
+            $('.form-group:has(#id_default_value)').show();
+            break;
+        case 'H':
+            input.replaceWith('<input type="time" class="form-control" value="' + value + '" ' + common_attrs + '>');
+            $('.form-group:has(#id_default_value)').show();
+            break;
+        case 'W':
+            input.replaceWith('<input type="datetime-local" class="form-control" value="' + value + '" ' + common_attrs + '>');
```

Comment:
We're not using type=date, type=time or type=datetime-local anywhere in pretix, as their UX is horrible in nearly all Desktop browsers. Could you instead try to use the date/time pickers we use everywhere already? :)